previous_code/gflownetenv.py

```diff
@@ -138,22 +146,22 @@ def state2obs(self, state=None):
         if state is None:
             state = self.state
         return state
-
+    #Do we actually need to convert a one-hot-encoding representation to a state in the pipeline ? This would be useless if the get_parent function outputs a manip_format instead of a one-hot-encoding format. (more coherent)
```

Comment:
Just making a not of where all it is used so far:
1. `unpack_terminal_states`: not sure I understand why
2. debug mode: might not be required in our implementation as we only get the ohe forms inside the proxy/policy model specifically.
3. setting the state in backward sampling
3.

Reply:
1. unpack_terminal_states is to remove some duplicates, etc ...
We will have to implement it eventually (maybe in querier.py)

Reply:
I see its implemented before feeding the data in the buffer. I think its implemented to filter the "done" states in the batch so that we can add just those to the buffer. Do you agree?

Reply:
1. unpack_terminal_states, not implemented yet, to be added eventually for the buffer
2. our get_parents output the parents in the consistent "manip" format, so no need of policy2manip (ie obs2state) so far